torchvision/prototype/transforms/_transform.py

```diff
@@ -0,0 +1,53 @@
+from typing import Any, Dict, Optional
+
+from torch import nn
+from torchvision.prototype.utils._internal import apply_recursively
+
+from .functional._utils import Dispatcher
+
+
+class Transform(nn.Module):
+    _DISPATCHER: Optional[Dispatcher] = None
+
+    def get_params(self, sample: Any) -> Dict[str, Any]:
+        return dict()
+
+    def _supports(self, obj: Any) -> bool:
+        if not self._DISPATCHER:
+            raise NotImplementedError()
+
+        return obj in self._DISPATCHER
+
+    def _dispatch(self, input: Any, params: Dict[str, Any]) -> Any:
+        if not self._DISPATCHER:
+            raise NotImplementedError()
+
+        return self._DISPATCHER(input, **params)
+
+    def _transform_recursively(self, sample: Any, params: Dict[str, Any]) -> Any:
+        def transform(input: Any) -> Any:
+            if not self._supports(input):
+                return sample
+
+            return self._dispatch(input, params)
+
+        return apply_recursively(transform, sample)
+
+    def forward(self, *inputs: Any, params: Optional[Dict[str, Any]] = None) -> Any:
+        sample = inputs if len(inputs) > 1 else inputs[0]
```

Comment:
Due to the `*inputs`, everything gets wrapped into a tuple. If we didn't have this line and you would get something like

```py
input = ...
transform = tranforms.Identity()

output = transform(input)

assert (input,) == output
```

We only have to perform this if we detect single inputs, because the behavior is already what you expect if you pass multiple inputs:

```py
image = ...
bounding_box = ...

transform = tranforms.Identity()

image2, bounding_box2 = transform(image, bounding_box)
```